test/converter.js

```diff
@@ -659,3 +697,45 @@ test('map switch schemas to unions', async function (test) {
   convert(context, parentType);
   await testConversion(test, childType, 'Child', expectedType, context);
 });
+
+test('definitions refs', async function (test) {
+  const type = {
+    id: 'Ref',
+    type: 'object',
+    definitions: {
+      otherType: {
+        type: 'object',
+        properties: {
+          attribute: {
+            type: 'string'
+          }
+        }
+      }
+    },
+    properties: {
+      attribute: {
+        $ref: '#/definitions/otherType'
+      }
+    }
+  };
+
+  const expectedType = `
+  type DefinitionOtherType {
```

Comment:
It's mostly about conflicts regarding naming. All type coming from the properties of the field will be unique (since two keys can't have the same name), and assuming that all schemas' ids will also be unique, there's no risk of conflicts there.

But how to ensure that we avoid the problem with definitions? Actually without the Definition prefix, it'll work (since the properties types are prefixed with the schema id), but I forgot the risk of conflicts between definitions of different schemas 😨. Perhaps a `Definition${Schema.id}${field}`? `${id}Definition` wouldn't work, since we could have an attribute named definition. I'll push the first solution, if you have a better idea, let me know!

Ideally, `Definition` should be configurable, but I don't think it's a priority for me, it could be done in a PR taking care of naming conflicts (you'd also have to ensure that the schema don't have the same id).